lib/cdo/contact_rollups.rb

```diff
@@ -203,9 +203,12 @@ def self.insert_from_dashboard_contacts
     log "Inserting teacher contacts and IP geo data from dashboard.users"
     PEGASUS_REPORTING_DB_WRITER.run "
     INSERT INTO #{PEGASUS_DB_NAME}.#{DEST_TABLE_NAME} (email, name, dashboard_user_id, roles, city, state, postal_code, country)
-    SELECT email COLLATE utf8_general_ci, name, users.id, 'Teacher', city, state, postal_code, country FROM #{DASHBOARD_DB_NAME}.users AS users
+    -- Use CONCAT+COALESCE to append 'Teacher' to any existing roles
+    SELECT users.email COLLATE utf8_general_ci, users.name, users.id, CONCAT(COALESCE(CONCAT(src.roles, ','), ''), 'Teacher'),
+    user_geos.city, user_geos.state, user_geos.postal_code, user_geos.country FROM #{DASHBOARD_DB_NAME}.users AS users
+    LEFT OUTER JOIN pegasus_development.contact_rollups_daily AS src ON src.email = users.email
     LEFT OUTER JOIN #{DASHBOARD_DB_NAME}.user_geos AS user_geos ON user_geos.user_id = users.id
-    WHERE users.user_type = 'teacher' AND LENGTH(email) > 0 AND user_geos.indexed_at IS NOT NULL
```

Comment:
It was overfiltering - we want the account in the rollup even if we don't have the geo data. Should have commented in the PR (better late than never!), not worth a comment in code.